File: MobileAppDev/app/lib/models/ticket.dart
class Ticket {
  final int id;
  final String problemType;
  final String description;
  final String customerName;
  final String billNumber;
  final String billDate;
  final int warehouseId;
  final String? warehouseName;
  final int status; // 0=Open, 1=InProgress, 2=Resolved, 3=Closed
  final String createdDate;
  final String? creatorId;
  final int commentCount;
  final String? lastCommentDate;

  // Local UI state (not from JSON directly)
  bool hasNewActivity;

  Ticket({
    required this.id,
    required this.problemType,
    required this.description,
    required this.customerName,
    required this.billNumber,
    required this.billDate,
    required this.warehouseId,
    this.warehouseName,
    required this.status,
    required this.createdDate,
    this.creatorId,
    this.commentCount = 0,
    this.lastCommentDate,
    this.hasNewActivity = false,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) {
    return Ticket(
      id: json['id'] ?? json['Id'],
      problemType: json['problemType'] ?? json['ProblemType'] ?? 'Unknown',
      description: json['description'] ?? json['Description'] ?? '',
      customerName: json['customerName'] ?? json['CustomerName'] ?? '',
      billNumber: json['billNumber'] ?? json['BillNumber'] ?? '',
      billDate: json['billDate'] ?? json['BillDate'] ?? '',
      warehouseId: json['warehouseId'] ?? json['WarehouseId'] ?? 0,
      warehouseName: json['warehouseName'] ?? json['WarehouseName'],
      status: json['status'] ?? json['Status'] ?? 0,
      createdDate: json['createdDate'] ?? json['CreatedDate'] ?? '',
      creatorId: json['creatorId'] ?? json['CreatorId'],
      commentCount: json['commentCount'] ?? json['CommentCount'] ?? 0,
      lastCommentDate: json['lastCommentDate'] ?? json['LastCommentDate'],
    );
  }

  String get statusText {
    switch (status) {
      case 0: return 'Open';
      case 1: return 'In Progress';
      case 2: return 'Resolved';
      case 3: return 'Closed';
      default: return 'Unknown';
    }
  }
}

File: iOS/TicketingSystem/Models/Ticket.swift
import Foundation

struct Ticket: Identifiable, Codable {
    let id: Int
    let problemType: String
    let description: String
    let customerName: String
    let billNumber: String
    let billDate: String
    let warehouseId: Int
    let warehouseName: String?
    let status: Int
    let createdDate: String
    let creatorId: String?
    let commentCount: Int
    let lastCommentDate: String?
    
    // Status enum for UI logic
    enum Status: Int {
        case open = 0
        case inProgress = 1
        case resolved = 2
        case closed = 3
        case unknown = -1
        
        var title: String {
            switch self {
            case .open: return "مفتوح"
            case .inProgress: return "قيد المعالجة"
            case .resolved: return "محلول"
            case .closed: return "مغلق"
            case .unknown: return "غير معروف"
            }
        }
    }
    
    var statusEnum: Status {
        return Status(rawValue: status) ?? .unknown
    }
}
